ghostfolio/ghostfolio
# feat: set up husky and pre commit hook for linting and format check

Fixes #3866 

I noticed that npm run lint will only run linting on the "api" project.
So i modified the command to make sure linting runs on all project.
The format check will only run on uncommitted files, this flag is sadly not available for linting. But due to nx caching i don't see a huge impact here, the first commit might take a bit longer in case nx has nothing cached so far.
The linting command has a `--quiet` flag which is required, as warnings will make the command exit with status code 1 too.
We can remove that flag once everything is tidy again.

## How to test:
Fail on format issues:
- Disable prettier in your IDE settings
- Modify a file in a way that formatting is definitely incorrect
- Commit that file and it should fail
- 
Fail on lint issues:
- Set any rule to "error" that is currently set to "warn", e.g. `@typescript-eslint/member-ordering`
- Commit that change and it should fail
